File: 0435-non-overlapping-intervals/0435-non-overlapping-intervals.cpp
class Solution {
public:
    int eraseOverlapIntervals(vector<vector<int>>& intervals) {
        int end = intervals.size();
        if (end==1){
            return 0;
        }
        sort(intervals.begin(), intervals.end());
        int res = 0;
        int prev_tail = intervals[0][1];
        for (int i = 1; i < end; i++){
            if (prev_tail <= intervals[i][0]){
                prev_tail = intervals[i][1];
                continue;
            }
            prev_tail = min(prev_tail, intervals[i][1]);
            res += 1;
        }
        return res;
    }
};

File: 0435-non-overlapping-intervals/0435-non-overlapping-intervals.py
class Solution:
    def eraseOverlapIntervals(self, intervals: List[List[int]]) -> int:
        intervals.sort()
        res = 0
        prev_tail = intervals[0][1]
        for start, end in intervals[1:]:
            if prev_tail <= start:
                prev_tail = end
                continue

            prev_tail = min(prev_tail,end)
            res += 1
        return res
            